src/lib/storage/proposal_settings.rb

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2015] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require_relative "./disk_size"
+require "pp"
+
+module Yast
+  module Storage
+    #
+    # User-configurable settings for the storage proposal.
+    # Those are settings the user can change in the UI.
+    #
+    class ProposalUserSettings
+      include Yast::Logger
+
+      attr_accessor :use_lvm, :encrypt_volume_group
+      attr_accessor :root_filesystem_type, :use_snapshots
+      attr_accessor :use_separate_home, :home_filesystem_type
+      attr_accessor :enlarge_swap_for_suspend
+
+      def initialize
+        @use_lvm                  = false
+        @encrypt_volume_group     = false
+        @root_filesystem_type     = ::Storage::BTRFS
+        @use_snapshots            = true
+        @use_separate_home        = true
+        @home_filesystem_type     = ::Storage::XFS
+        @enlarge_swap_for_suspend = false
+      end
+    end
```

Comment:
I think it make sense to consider using Struct class as this is mainly structure, not class with methods. see http://ruby-doc.org/core-2.2.0/Struct.html